### DISCUSSION TITLE
Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com:6543`

### DISCUSSION THREAD

**Discussion by GitVex** [prisma/prisma#24881] [Q&A] [answered]
### Question

I'm running into a problem with connecting to my supabase postgresql database. I'm simply trying to retrieve some items from it, but get this error:

```
 X PrismaClientInitializationError: 
Invalid `prisma.track.findMany()` invocation:


Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com:6543`

Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com:6543`.
    at _n.handleRequestError (C:\Users\paulk\WebstormProjects\BardicInspiration\node_modules\@prisma\client\runtime\library.js:121:8049)
    at _n.handleAndLogRequestError (C:\Users\paulk\WebstormProjects\BardicInspiration\node_modules\@prisma\client\runtime\library.js:121:7057)
    at _n.request (C:\Users\paulk\WebstormProjects\BardicInspiration\node_modules\@prisma\client\runtime\library.js:121:6741)
    at async l (C:\Users\paulk\WebstormProjects\BardicInspiration\node_modules\@prisma\client\runtime\library.js:130:9355)
    at async handle (webpack-internal:///(api)/./pages/api/viewer/list.ts:16:20) {
  clientVersion: '5.17.0',
  errorCode: undefined,
  page: '/api/viewer/list'
}
```

I've tried multiple things, mostly covered in https://github.com/orgs/supabase/discussions/4161 and https://github.com/prisma/prisma/discussions/23356.
Most notably:
- I've ensured that my connection string is correct through conneting to my database with Jetbrains DataGrip and DataSpell, so all the fields are correct, and I am sure that my database is running at `aws-0-eu-central-1.pooler.supabase.com:6543`.
- Since I've seen it around a couple different discussions, switching networks and disabling/enabing VPN's also didn't resolve the issue.
- An obligatory system restart also didn't fix it.
- There are also no special characters that need any percent-substitution in the username or password

WHAT THE ISSUE MIGHT BE: I had to rotate the database password, and after doing so the connection did not work anymore, neither with the prisma connection, nor with DataGrip or -Spell. After rebooting my system the later two worked with the correct password, but prisma still doesn't. Could there be some case where I need to reset a cache for the database credentials?

I'll provide other details below.

### Information about Prisma Schema, Client Queries and Environment (optional)

.env.local:
```.env
# Connect to Supabase via connection pooling with Supavisor.
DATABASE_URL="postgresql://[USER]:[PASSWORD]@aws-0-eu-central-1.pooler.supabase.com:6543/postgres?pgbouncer=true"

# Direct connection to the database. Used for migrations.
DIRECT_URL="postgres://[USER]:[PASSWORD]@aws-0-eu-central-1.pooler.supabase.com:5432/postgres"
```

schema.prisma:
```typescript
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider          = "postgresql"
  url               = env("DATABASE_URL")
  directUrl         = env("DIRECT_URL")
}

// .... Table declarations
```

just for completeness this is also the query:
```typescript
import { NextApiRequest, NextApiResponse } from 'next';
import { prisma } from '../../../utils/prismaClientProvider';
import TItem from '../../../components/Viewer/types/TItem';

export default async function handle(req: NextApiRequest, res: NextApiResponse) {
    const { page = 0, pageSize = 10, excludeIds = [] } = req.query;

    // Pagination ...

    const result = await prisma.track.findMany({
        where: {
            track_id: {
                notIn: excludeIdsArray,
            },
        },
        orderBy: {
            created_at: 'desc',
        },
        include: {
            tags: true,
            artist: true,
        },
        skip: pageNumber * pageSizeNumber,
        take: pageSizeNumber,
    });
     
// Post Processing ...

    // Prepare pagination info
    const totalRecords = await prisma.track.count();
    const totalPages = Math.ceil(totalRecords / pageSizeNumber);

    const pageObj = {
        data: result as TItem[],
        pagination: {
            totalRecords,
            totalPages,
            currentPage: pageNumber,
            pageSize: pageSizeNumber,
        },
    };

    res.status(200).json(pageObj);
}
```

My project also ran on an older version of prisma, but as I tried to fix my problem I updated to prisma 5, which didn't resolve the problem.
So I'm running:
```npm
 @prisma/client@5.17.0
 prisma@5.17.0
 next@13.5.1

```

**Comment by ludralph** [comment]
Hi @GitVex 👋 

Thank you for raising this question.

Do you have any errors when you perform migrations or do a `db push` command? Can you also confirm that the database instance on Supabase is not paused?

**Reply by GitVex** [reply]
Hey! Thank you for the quick reply.

I can confirm that the supabase is not paused.
I tried to do a `db push` (as well as `db pull`), and got the error that prisma doesnt seem to find the environment variables:

```
npx prisma db pull
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database
Error: Prisma schema validation - (get-config wasm)
Error code: P1012
error: Environment variable not found: DIRECT_URL.
  -->  prisma\schema.prisma:8
   | 
 7 |   url               = env("DATABASE_URL")
 8 |   directUrl         = env("DIRECT_URL")
   | 

Validation Error Count: 1
[Context: getConfig]

Prisma CLI Version : 5.17.0
```

Which is weird, as it does find the variables when I run the app with `npm run dev`.

**Reply by GitVex** [reply, marked as answer]
Well ... this is awkward ... but after reseting the Database password again it now works again in all cases ...

I don't know why or how ... but I guess it's fixed now

**Reply by HosseinEnsafi** [reply]
it was probably because your password had @ or # char in it . in this case the password is confused with part of url connection

**Comment by akhtaralikhandev** [comment]
I faced this and I added this to the url aws-0-us-west-1.pooler.supabase.com:5432/postgres?statement_timeout=60000" timeout and it worked

**Reply by xd-elon** [reply]
thanks

**Reply by Sh0utaa** [reply]
wow are you kidding me, thanks this worked

**Reply by chukwudobemicah** [reply]
this is for the DIRECT_URL and not the DATABASE_URL right?

**Reply by taozhiyuzhuo** [reply]
you used port 5432 for DATABASE_URL not 6543?

**Reply by deva3xd** [reply]
> you used port 5432 for DATABASE_URL not 6543?

man, this is for direct url not database url

**Reply by taozhiyuzhuo** [reply]
> > you used port 5432 for DATABASE_URL not 6543?
> 
> man, this is for direct url not database url

Actually, the DATABASE_URL can also use the direct connection URL

**Comment by adeoluwa** [comment]
creating another project solves the issue for me.

**Comment by ScottishRoss** [comment]
For anyone else that encounters this, refreshing the page a few times seems to work for me. This is on my local dev server, so I certainly hope this doesn't happen in Prod.

**Comment by olivierlesnicki** [comment]
I confirm this was the issue for me too.
Had special characters in the password.

**Comment by krngd2** [comment]
I have the same issue, restarting the local server works and then again disconnects in few mins

**Reply by krngd2** [reply]
I solved it by just disabling the VPN

**Comment by KevinAshford** [comment]
try restarting the db server, if not resolved at once do twice, it will solve the issue

**Reply by krngd2** [reply]
Just tried this, did not work

**Comment by yafet-dev** [comment]
I also getting this error again and again, it works after some time, it's just not stable at all and this issue shoudn't be resolved 
<img width="1716" height="67" alt="image" src="https://github.com/user-attachments/assets/5191f61a-a2e1-48d4-962e-921f888dc749" />

**Reply by krngd2** [reply]
In my case my VPN was on, check it on your side

**Comment by rozenonline** [comment]
For me, I just restarted supabase project, and it works now

**Comment by ryanyu104** [comment]
same issue for me too, It doesn't work after I reset the db password.
